src/liveEditor/listeners/mouseClick.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    cleanIndividualFieldResidual,
+    handleIndividualFields,
+} from "../utils/handleIndividualFields";
+
+import {
+    getCsDataOfElement,
+    getDOMEditStack,
+} from "../utils/getCsDataOfElement";
+
+import { appendFocusedToolbar } from "../generators/generateToolbar";
+
+import { addFocusOverlay } from "../generators/generateOverlay";
+
+import liveEditorPostMessage from "../utils/liveEditorPostMessage";
+
+import { LiveEditorPostMessageEvents } from "../utils/types/postMessage.types";
+
+import EventListenerHandlerParams from "./params";
+import VisualEditorGlobalState from "../globals";
+
+export interface HandleMouseClickParams
+    extends Omit<EventListenerHandlerParams, "eventDetails" | "customCursor"> {}
+
+interface AddFocusOverlayParams
+    extends Pick<
+        EventListenerHandlerParams,
+        "overlayWrapper" | "resizeObserver"
+    > {
+    editableElement: Element;
+}
+
+interface AddFocusedToolbarParams
+    extends Pick<
+        EventListenerHandlerParams,
+        "eventDetails" | "focusedToolbar"
+    > {}
+
+function addOverlay(params: AddFocusOverlayParams) {
+    if (!params.overlayWrapper || !params.editableElement) return;
+
+    addFocusOverlay(params.editableElement, params.overlayWrapper);
+    params.resizeObserver.observe(params.editableElement);
+}
+
+function addFocusedToolbar(params: AddFocusedToolbarParams) {
+    const { editableElement } = params.eventDetails;
+
+    if (!editableElement || !params.focusedToolbar) return;
+
+    // Don"t append again if already present
+    if (
+        VisualEditorGlobalState.value.previousSelectedEditableDOM &&
+        VisualEditorGlobalState.value.previousSelectedEditableDOM ===
+            editableElement
+    ) {
+        return;
+    }
+
+    appendFocusedToolbar(params.eventDetails, params.focusedToolbar);
+}
+
+async function handleMouseClick(params: HandleMouseClickParams): Promise<void> {
```

Comment:
We can rename this function better. Let's describe what it does when we click something.

Reply:
Is `handleEditorInteraction` okay ?